### PR TITLE
fix oauth issues with incorrect values

### PR DESCRIPTION
**Pull request by DanielPalafox**
https://trello.com/c/znWvsONw/6960-qa-oauth-issues-with-responses-to-calls-made-with-incorrect-clientids-and-scopes-or-unregistered-redirecturis

### CODE REVIEW

**Comment by DanielPalafox** [comment]
[[QA] OAuth, issues with responses to calls made with incorrect client_ids and scopes, or unregistered redirect_uri's](https://trello.com/c/znWvsONw/6960-qa-oauth-issues-with-responses-to-calls-made-with-incorrect-clientids-and-scopes-or-unregistered-redirecturis)

File: src/app/types/request-info-form.endpoint.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export interface RequestInfoForm {
   nonce: string
   clientHavePersistentTokens: boolean
   scopesAsString: string
-  error: 'login_required' | 'interaction_required'
```
Comment (leomendoza123) [inline src/app/types/request-info-form.endpoint.ts:23]: are the more errors here than just 'login_required' | 'interaction_required', can't we just list those as well?

File: src/app/guards/authorize.guard.ts [hunk 2]
```diff
@@ -54,13 +54,19 @@ export class AuthorizeGuard implements CanActivateChild {
     return this._user.getUserSession(queryParams).pipe(
       map((x) => x.oauthSession),
       map((session) => {
-        if (session.forceLogin || !oauthSessionUserIsLoggedIn(session)) {
-          return this.redirectToLoginPage(queryParams)
+        if (session.redirectUrl && session.responseType && session.redirectUrl.includes(session.responseType)) {
+          window.location.href = session.redirectUrl
```
Comment (leomendoza123) [inline src/app/guards/authorize.guard.ts:62]: please use the window service

File: src/app/guards/authorize.guard.ts [hunk 3]
```diff
@@ -54,13 +54,26 @@ export class AuthorizeGuard implements CanActivateChild {
     return this._user.getUserSession(queryParams).pipe(
       map((x) => x.oauthSession),
       map((session) => {
-        if (session.forceLogin || !oauthSessionUserIsLoggedIn(session)) {
-          return this.redirectToLoginPage(queryParams)
+        if (
+          session.redirectUrl &&
+          session.responseType &&
+          session.redirectUrl.includes(session.responseType)
```
Comment (leomendoza123) [inline src/app/guards/authorize.guard.ts:60]: do we now all the possible values of `responseType`? Is it possible to add that to the types

**Review by leomendoza123 — CHANGES_REQUESTED** [review]

File: src/app/guards/authorize.guard.ts (hunk 3 above)
Comment (DanielPalafox) [reply src/app/guards/authorize.guard.ts:60]: Ok, I will add them

File: src/app/types/request-info-form.endpoint.ts (hunk 1 above)
Comment (DanielPalafox) [reply src/app/types/request-info-form.endpoint.ts:23]: Yes, there are more. Ok, I will list them all

File: src/app/types/request-info-form.endpoint.ts [hunk 4]
```diff
@@ -31,3 +31,13 @@ export interface Scope {
   description: string
   longDescription: string
 }
+
+enum Errors {
```
Comment (leomendoza123) [inline src/app/types/request-info-form.endpoint.ts:35]: for this purpose, a type is required not an enum. 

Sinde the error itself will not be a enum

**Review by leomendoza123 — CHANGES_REQUESTED** [review]

File: src/app/types/request-info-form.endpoint.ts (hunk 4 above)
Comment (DanielPalafox) [reply src/app/types/request-info-form.endpoint.ts:35]: Yes, you are right. I will fix it

**Review by leomendoza123 — APPROVED** [review]

**Pull request merged**